src/thunderbots/software/ai/world/team.h

```diff
@@ -20,22 +21,62 @@ class Team
    public:
     /**
      * Create a new team
+     *
+     * @param robot_expiry_buffer_milliseconds The number of milliseconds a robot must not
+     * have been updated for before it is removed from the team
      */
-    explicit Team();
+    explicit Team(const std::chrono::milliseconds robot_expiry_buffer_milliseconds);
 
     /**
      * Updates this team with new robots.
      *
      * @param team_robots the new robots for this team
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateRobots(const std::vector<Robot>& team_robots,
+                      const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Updates this team with new data from the given team object
+     *
+     * @param new_team_data A team with the new team data
+     * @param timestamp The timestamp at which this update is taking place. The timestamp
+     * must be >= the last update timestamp of the robots on the team
+     */
+    void updateState(const Team& new_team_data,
+                     const std::chrono::steady_clock::time_point timestamp);
+
+    /**
+     * Updates the Team's state to be its predicted state at the given timestamp.
+     * The timestamp must be >= the last update timestamp of the robots on the team
+     *
+     * @param timestamp The timestamp at which to update the team's state to. Must
+     * be >= the last update timestamp of the robots on the team
+     */
+    void updateStateToPredictedState(
+        const std::chrono::steady_clock::time_point timestamp);
```

Comment:
I'm not sure of the context this will be used in, but would it make more sense to have it return a new `Team`, thus allowing this team to retain the true state as last seen? (Just a suggestion)

Reply:
The context would be when the AI node ticks and no new data was received, it would call this function with the current time rather than the other `updateState` functions that accept data. This way the AI will be operating with en estimate of what the most up to date state of the world is. It will continue to update and use the estimated state until new data is received again.

I think it makes sense to leave it as it is, since it matches the other `update` functions. We _could_ make all `update` functions return a new object, but I would need a better idea of the pros/cons.

Reply:
Leaving it as it is.